src/JsRoot/JsRootGraph2D.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from 'react';
+import Skeleton from '@mui/material/Skeleton';
+import { useJSROOT } from './JsRootService';
+import { useVisible } from 'react-hooks-visible';
+import { Page2D } from './GraphData';
+
+export function JsRootGraph2D(props: Page2D) {
+	const { JSROOT } = useJSROOT();
+	const [containerEl, visible] = useVisible<HTMLDivElement>();
+
+	const [obj, setObj] = useState(undefined);
+	const [drawn, setDrawn] = useState(false);
+	const [isVisible, setIsVisible] = useState(false);
+
+	useEffect(() => {
+		setIsVisible(visible > 0.3);
```

Comment:
what is that 0.3 constant ?

Reply:
It's an arbitrary constant that determines if the graph should be loaded based on how much of the graph is visible.

Reply:
ok